src/watchmaker/__init__.py

```diff
@@ -301,6 +304,12 @@ def _get_config(self):
             self.log.critical(msg)
             raise WatchmakerException(msg)
 
+        if config_version_specifier and not check_version(
+                watchmaker.__version__, config_version_specifier):
+            msg = 'Watchmaker is not compatible with the config file'
+            self.log.critical(msg)
+            raise WatchmakerException(msg)
+
         # Merge the config data, preserving the listed order of workers
         config = collections.OrderedDict()
         for worker in config_system + config_all:
```

Comment:
I'm not sure if it's worth it, but somewhere in here we _could_ check for string `'None'` values, force them to `None`, and log/raise a [`DeprecationWarning`](https://docs.python.org/3/library/warnings.html#warning-categories)...

Reply:
This should not be too hard. I'll get it in.

Reply:
Take a look at my approach to this and see what you think.

Reply:
My approach is to centralize the handling of `None`. Since it is centralized, it will be easy to switch to raising an error and then phasing out.

Reply:
Good enough for now, until we get a real config spec/parser